templates/apis/types.go.tpl

```diff
@@ -14,3 +14,15 @@ import (
 
 {{ template "type_def" $typeDef }}
 {{- end -}}
+{{- if .HasSecret }}
+// SecretReference has enough information to retrieve secret
+// in any namespace.
+type SecretReference struct {
+    // Namespace defines the space within which the secret name must be unique.
+    Namespace string `json:"namespace,omitempty"`
+    // Name of secret in a given namespace.
+    Name string `json:"name,omitempty"`
+    // Key of the secret.
+    Key string `json:"key,omitempty"`
+}
```

Comment:
this should be in the ACK core types (i.e. github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1). See my comment from above.

There should not be a need for a .HasSecret boolean field of the CRD.